src/routes/userRoutes.test.js

```diff
@@ -0,0 +1,93 @@
+jest.mock('mailgun-js')
+jest.mock('../../lib/users')
+jest.mock('../../sequelize/db')
+
+const db = require('../../sequelize/db')
+const {resetPassword} = require('./userRoutes')
+const {sendPasswordResetEmail} = require('../../lib/users')
+
+const mockJson = jest.fn()
+
+const res = {
+  status: jest.fn().mockReturnValue({json: mockJson})
+}
+
+describe('Testing user routes', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+  test('should send 400 error if req body does not have email', async () => {
+    const req = {
+      body: {}
+    }
+
+    await resetPassword(req, res)
+    expect(res.status.mock.calls[0][0]).toEqual(400)
+    expect(mockJson.mock.calls[0][0]).toEqual({error: {message: "invalid input"}})
+  })
+
+  test('Should send 400 error if user account does not exist', async () => {
+    db.getModels = jest.fn().mockReturnValue({
+      Accounts: {
+        findOne: jest.fn() 
+      } 
+    }) 
+
+   const req = {
+     body: {
+       email: 'hello@world.com',
+     } 
+   } 
+
+    await resetPassword(req, res)
+    expect(res.status.mock.calls[0][0]).toEqual(400)
+    expect(mockJson.mock.calls[0][0]).toEqual({error: {message: "account does not exist"}})
+  })
+
+  test('should send 500 error if send password throws error', async () => {
+    db.getModels = jest.fn().mockReturnValue({
+      Accounts: {
+        findOne: jest.fn().mockReturnValue({
+          where: {
+            email: jest.fn()
+          }
+        })
+      }
+    })
+
+    const req = {
+      body: {
+        email: 'hello@world.com'
+      }
+    }
+
+    await resetPassword(req, res)
+    expect(res.status.mock.calls[0][0]).toEqual(500)
```

Comment:
![image](https://user-images.githubusercontent.com/61715204/84118847-84d69000-aa6e-11ea-8b7c-b5968d8666d7.png)
It recieved `200` which means it didn't throw error
![image](https://user-images.githubusercontent.com/61715204/84118955-aafc3000-aa6e-11ea-8903-3dd736968362.png)
1. `resetPassword` doesn't throw any error so the description is wrong. It sends json object that has error property.
2. It sends error when `email` is falsey. Which means we should put no email value in `req.body` in order to recieve 400 error

Reply:
*In `userRoutes.js`
![image](https://user-images.githubusercontent.com/61715204/84119667-a8e6a100-aa6f-11ea-8162-a9503a40516f.png)
`resetPassword` sends `500` status when `sendPasswordResetEmail` throws error
*In `users.js`
![image](https://user-images.githubusercontent.com/61715204/84119848-f400b400-aa6f-11ea-93bb-0d9d643e155a.png)
`users.sendpasswordResetEmail` may throw error when `email.sendPasswordResetEmail` throws error.
(email is required from `mailer.js`)
*In mailer.js
![image](https://user-images.githubusercontent.com/61715204/84119576-86ed1e80-aa6f-11ea-8747-411ffc7773cf.png)
`mgModule.sendPasswordResetEmail` will throw error when `mg.messages().send(data)` fails. `data` has `to` property whose value is `receiver` which is email address. So I assume you can make it throw error by passing a wrong email address

Reply:
Or you can simply mock `users.sendPasswordResetEmail` then do `sendPasswordResetEmail.mockImplement(() => { throw new Error('error') })`